### PR TITLE
Update BWC Test Version and Enhance Code Coverage

### PR DESCRIPTION
**Pull request by kaituo**
### Description
This PR
* Updated the BWC test version to 2.16 in the build.gradle file.
* Added new integration and unit tests to enhance code coverage.
* Fixed an issue where integration tests were not being counted towards coverage.

Testing Performed:
* Executed gradle build to ensure successful build and integration.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by kaituo** [comment]
Not sure why WhiteSource Security Check failed. It shows it is scanning but never finishes. Will leave it as it is.

File: build-tools/coverage.gradle [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+ */
+
+apply plugin: 'jacoco'
+
+jacoco {
+    toolVersion = "0.8.10"
+}
+
+/**
+ * This code sets up coverage reporting manually for the k-NN plugin tests. This is complicated because:
```
Comment (amitgalitz) [inline build-tools/coverage.gradle:13]: nit: change to AD plugin tests

Comment (kaituo) [reply build-tools/coverage.gradle:13]: changed

File: src/main/java/org/opensearch/forecast/settings/ForecastEnabledSetting.java [hunk 2]
```diff
@@ -27,7 +27,8 @@ public class ForecastEnabledSetting extends DynamicNumericSetting {
      */
     public static final String FORECAST_ENABLED = "plugins.forecast.enabled";
 
-    public static final boolean enabled = false;
+    // TODO: remove the field when releasing forecasting
+    public static boolean enabled = false;
```
Comment (amitgalitz) [inline src/main/java/org/opensearch/forecast/settings/ForecastEnabledSetting.java:31]: is this boolean being used?

Comment (kaituo) [reply src/main/java/org/opensearch/forecast/settings/ForecastEnabledSetting.java:31]: it is not being used. Removed.

**Review by amitgalitz — APPROVED** [review]

**Pull request merged**